### PR TITLE
Decompile fixes for ProtoChunk and DataPaletteBlock

### PR DESCRIPTION
**Pull request by Spottedleaf**

### CODE REVIEW

File: Spigot-Server-Patches/0003-MC-Dev-fixes.patch [hunk 1]
```diff
@@ -71,8 +71,39 @@ index 650ef14758..35aea4829f 100644
...
+diff --git a/src/main/java/net/minecraft/server/DataPaletteBlock.java b/src/main/java/net/minecraft/server/DataPaletteBlock.java
+index 20b1bc1b4..e9c59b7d7 100644
+--- a/src/main/java/net/minecraft/server/DataPaletteBlock.java
++++ b/src/main/java/net/minecraft/server/DataPaletteBlock.java
+@@ -85,7 +85,7 @@ public class DataPaletteBlock<T> implements DataPaletteExpandable<T> {
+             Object object = datapalette.a(databits.a(j));
+ 
+             if (object != null) {
+-                this.setBlockIndex(j, object);
++                this.setBlockIndex(j, (T) object); // Paper decompile fix
```
Comment (stonar96) [inline Spigot-Server-Patches/0003-MC-Dev-fixes.patch]: Why not change `Object object` to `T object` instead?

File: Spigot-Server-Patches/0003-MC-Dev-fixes.patch [hunk 2]
```diff
@@ -71,8 +71,39 @@ index 650ef14758..35aea4829f 100644
...
+             }
+         }
+ 
+@@ -113,7 +113,7 @@ public class DataPaletteBlock<T> implements DataPaletteExpandable<T> {
+     protected T a(int i) {
+         Object object = this.h.a(this.a.a(i));
+ 
+-        return object == null ? this.g : object;
++        return object == null ? this.g : (T) object; // Paper - decompile fix
```
Comment (stonar96) [inline Spigot-Server-Patches/0003-MC-Dev-fixes.patch]: Same here.

File: Spigot-Server-Patches/0003-MC-Dev-fixes.patch (hunk 1 above)
Comment (Spottedleaf) [reply Spigot-Server-Patches/0003-MC-Dev-fixes.patch]: will do

File: Spigot-Server-Patches/0003-MC-Dev-fixes.patch (hunk 2 above)
Comment (Spottedleaf) [reply Spigot-Server-Patches/0003-MC-Dev-fixes.patch]: will do

File: Spigot-Server-Patches/0003-MC-Dev-fixes.patch [hunk 3]
```diff
@@ -229,8 +260,57 @@ index 86585a13ff..f52b9c6f0a 100644
...
+diff --git a/src/main/java/net/minecraft/server/ProtoChunk.java b/src/main/java/net/minecraft/server/ProtoChunk.java
+index 824e68b50..d0bb115eb 100644
+--- a/src/main/java/net/minecraft/server/ProtoChunk.java
++++ b/src/main/java/net/minecraft/server/ProtoChunk.java
+@@ -65,7 +65,7 @@ public class ProtoChunk implements IChunkAccess {
+         this.b = chunkcoordintpair;
+         this.p = chunkconverter;
+         Predicate predicate = (block) -> {
+-            return block == null || block.getBlockData().isAir();
++            return block == null || ((Block)block).getBlockData().isAir(); // Paper - decompile fix
```
Comment (stonar96) [inline Spigot-Server-Patches/0003-MC-Dev-fixes.patch]: Add space after cast. (The `predicate` variable is reused. (Just a note for me.))

File: Spigot-Server-Patches/0003-MC-Dev-fixes.patch [hunk 4]
```diff
@@ -229,8 +260,57 @@ index 86585a13ff..f52b9c6f0a 100644
...
+         RegistryBlocks registryblocks = Block.REGISTRY;
+ 
+@@ -325,8 +325,8 @@ public class ProtoChunk implements IChunkAccess {
+     }
+ 
+     private HeightMap c(HeightMap.Type heightmap_type) {
+-        return (HeightMap) this.f.computeIfAbsent(heightmap_type, (heightmap_type) -> {
+-            HeightMap heightmap = new HeightMap(this, heightmap_type);
++        return (HeightMap) this.f.computeIfAbsent(heightmap_type, (heightmap_type2) -> { // Paper - decompile fix
++            HeightMap heightmap = new HeightMap(this, heightmap_type2); // Paper - decompile fix
```
Comment (stonar96) [inline Spigot-Server-Patches/0003-MC-Dev-fixes.patch]: Change 2 to 1 to be consistent.

File: Spigot-Server-Patches/0003-MC-Dev-fixes.patch (hunk 3 above)
Comment (Spottedleaf) [reply Spigot-Server-Patches/0003-MC-Dev-fixes.patch]: will do

File: Spigot-Server-Patches/0003-MC-Dev-fixes.patch (hunk 4 above)
Comment (Spottedleaf) [reply Spigot-Server-Patches/0003-MC-Dev-fixes.patch]: will do

**Comment by stonar96** [comment]
That was everything. Thank you for helping me out with the lambdas!

File: Spigot-Server-Patches/0003-MC-Dev-fixes.patch [hunk 5]
```diff
@@ -71,8 +71,39 @@ index 650ef14758..35aea4829f 100644
...
+diff --git a/src/main/java/net/minecraft/server/DataPaletteBlock.java b/src/main/java/net/minecraft/server/DataPaletteBlock.java
+index 20b1bc1b4..590794d85 100644
+--- a/src/main/java/net/minecraft/server/DataPaletteBlock.java
++++ b/src/main/java/net/minecraft/server/DataPaletteBlock.java
+@@ -82,7 +82,7 @@ public class DataPaletteBlock<T> implements DataPaletteExpandable<T> {
+         int j;
+ 
+         for (j = 0; j < databits.b(); ++j) {
+-            Object object = datapalette.a(databits.a(j));
++            T object = (T) datapalette.a(databits.a(j)); // Paper - decompile fix
```
Comment (stonar96) [inline Spigot-Server-Patches/0003-MC-Dev-fixes.patch]: Sorry. Can you do one more please? Change `DataPalette datapalette` to `DataPalette<T> datapalette` and remove the cast.

Comment (Spottedleaf) [reply Spigot-Server-Patches/0003-MC-Dev-fixes.patch]: ok

**Comment by stonar96** [comment]
Thanks again.

**Pull request merged**